src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,387 @@
+    </Box>
+  )
+}
+
+export type TitleProps = {
+  variant?: 'subtitle' | 'medium' | 'large' | ResponsiveValue<'subtitle' | 'medium' | 'large'>
+  // Check if we need responsive values for heading is so should we update as prop's type for Heading component?
+  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+} & PageHeaderProps
+
+const Title: React.FC<React.PropsWithChildren<TitleProps>> = ({
+  children,
+  sx = {},
+  variant = 'medium',
+  hidden = false,
+  as = 'h3',
+}) => {
+  const currentVariant = useResponsiveValue(variant, 'medium')
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    // TODO: clarify if we want rem for line height or not
+    <Heading
+      as={as}
+      sx={merge<BetterSystemStyleObject>(
+        {
+          fontSize: {
+            large: '2rem',
+            medium: '1.25rem',
+            subtitle: '1.25rem',
+          }[currentVariant],
+          lineHeight: {
+            large: '48px',
+            medium: '32px',
+            subtitle: '32px',
+          }[currentVariant],
+          fontWeight: {
+            large: '400',
+            medium: '400',
+            subtitle: '400',
+          }[currentVariant],
+          display: isHidden ? 'none' : 'flex',
+        },
+        sx,
+      )}
+    >
+      {children}
+    </Heading>
+  )
+}
+const TrailingVisual: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, hidden = false}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isHidden ? 'none' : 'flex',
+        },
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+const TrailingAction: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  hidden = hiddenOnNarrow,
+}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return <Box sx={merge<BetterSystemStyleObject>({display: isHidden ? 'none' : 'flex'}, sx)}>{children}</Box>
+}
+
+const Actions: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, hidden = false}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isHidden ? 'none' : 'flex',
+          flexDirection: 'row',
+          gap: get('space.2'),
+          flexGrow: '1',
+          justifyContent: 'right',
+        },
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+// PageHeader.Description: The description area of the header. Visible on all viewports
+const Description: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, hidden = false}) => {
+  const isHidden = useResponsiveValue(hidden, true)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isHidden ? 'none' : 'flex',
+          flexDirection: 'row',
+          alignItems: 'center',
+          gap: get('space.2'),
```

Comment:
I think we should use `get('stack.gap.condensed')` (from [UI stack tokens](https://primer.style/primitives/spacing#ui-stack))

cc @langermank